### PR TITLE
feat(*): Remove legacy comparison views code

### PR DESCRIPTION
**Pull request by grafakus**
### ✨ Description

**Related issue(s):** is a follow-up of https://github.com/grafana/explore-profiles/pull/119, is blocked by https://github.com/grafana/explore-profiles/pull/164

Now that the comparison view has been migrated to Scenes, we can delete the corresponding code.

### 📖 Summary of the changes

image

Deleting code! 🥳🥳

### 🧪 How to test?

- The build should pass

### CODE REVIEW

File: src/pages/ProfilesExplorerView/domain/variables/ServiceNameVariable/ServiceNameVariable.tsx [hunk 1]
```diff
@@ -8,13 +8,13 @@ import {
...
-import { buildServiceNameCascaderOptions } from '@shared/components/Toolbar/domain/useBuildServiceNameOptions';
 import { reportInteraction } from '@shared/domain/reportInteraction';
 import React, { useMemo } from 'react';
 import { lastValueFrom } from 'rxjs';
 
-import { PYROSCOPE_SERIES_DATA_SOURCE } from '../../infrastructure/pyroscope-data-sources';
-import { FiltersVariable } from './FiltersVariable/FiltersVariable';
+import { PYROSCOPE_SERIES_DATA_SOURCE } from '../../../infrastructure/pyroscope-data-sources';
+import { FiltersVariable } from '../FiltersVariable/FiltersVariable';
+import { buildServiceNameCascaderOptions } from './domain/useBuildServiceNameOptions';
```
Comment (grafakus) [inline src/pages/ProfilesExplorerView/domain/variables/ServiceNameVariable/ServiceNameVariable.tsx:17]: We brought this function here.

File: src/shared/domain/url-params/useQueryFromUrl.ts [hunk 2]
```diff
@@ -38,26 +30,15 @@ function useSetDefaultQuery(hasQuery: boolean, setQuery: (newQuery: string) => v
...
-function buildHook(targetTimeline: TargetTimeline) {
-  const paramName = PARAM_NAMES.get(targetTimeline);
-  if (paramName === undefined) {
-    throw new TypeError(`Undefined parameter name for "${targetTimeline}" timeline!`);
-  }
-
-  return function useQueryFromUrl(): [string, (newQuery: string) => void] {
-    const { searchParams, pushNewUrl } = useUrlSearchParams();
-    const query = searchParams.get(paramName) ?? '';
+export function useQueryFromUrl(): [string, (newQuery: string) => void] {
```
Comment (grafakus) [inline src/shared/domain/url-params/useQueryFromUrl.ts:33]: We just need one now.

File: e2e/fixtures/pages/ExploreProfilesPage.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { expect, type Page } from '@playwright/test';
 
-import { DEFAULT_EXPLORE_PROFILES_URL_PARAMS } from '../../config/constants';
```
Comment (grafakus) [inline e2e/fixtures/pages/ExploreProfilesPage.ts:3]: Changes here and there taken from https://github.com/grafana/explore-profiles/pull/136

File: e2e/config/constants.ts [hunk 4]
```diff
@@ -40,8 +45,18 @@ export const DEFAULT_EXPLORE_PROFILES_URL_PARAMS = ENV_VARS.E2E_BASE_URL.startsW
       'var-dataSource': 'grafanacloud-profiles-local-a',
       'var-serviceName': 'ride-sharing-app',
       'var-profileMetricId': 'process_cpu:cpu:nanoseconds:cpu:nanoseconds',
-      maxNodes: '16384',
+      // no max nodes here so that the app fetches user settings and we can also test settings changes
```
Comment (bryanhuhta) [inline e2e/config/constants.ts:48]: 👌

**Review by bryanhuhta — APPROVED** [review]
Love to see code get removed! Nice work

File: docs/sources/choose-a-view/index.md [hunk 5]
```diff
@@ -170,5 +170,5 @@ It normalizes the data by comparing the percentage of total time spent in each f
 This lets you compare two different queries that may have different total amounts of time spent in each function.
 
 Similar to a `git diff`, it takes the selected flame graphs and highlights the differences between them.
-<!-- from old screenshot of cpu usage profile Red represents an increase in CPU usage from the baseline to the comparison and green represents a decrease. -->
 
+<!-- from old screenshot of cpu usage profile Red represents an increase in CPU usage from the baseline to the comparison and green represents a decrease. -->
```
Comment (grafakus) [inline docs/sources/choose-a-view/index.md:174]: Auto-lint

File: docs/sources/determine-use-case/index.md [hunk 6]
```diff
@@ -59,5 +59,4 @@ After you locate the profile with a spike, select either **Profile types** view
 Illuminate performance issues across a number of dimensions when you are doing a proactive analysis.
 This could be a cost-cutting exercise or attempt to improve latency or memory usage in a non-incident scenario. This exercise could be across multiple services or within a service across profile types, labels, or a combination of the two.
 
-
```
Comment (grafakus) [inline docs/sources/determine-use-case/index.md:62]: Auto-lint

File: src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/components/SceneFunctionDetailsPanel/components/CodeContainer/infrastructure/useFetchVCSFile.ts [hunk 7]
```diff
@@ -21,7 +21,14 @@ type FetchResponse = {
   };
 };
 
-export function useFetchVCSFile({ enabled, dataSourceUid, repository, gitRef, localPath, rootPath }: FetchParams): FetchResponse {
+export function useFetchVCSFile({
```
Comment (grafakus) [inline src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/components/SceneFunctionDetailsPanel/components/CodeContainer/infrastructure/useFetchVCSFile.ts:24]: Auto-lint

**Pull request merged**